mkosi

```diff
@@ -982,6 +983,70 @@ def invoke_dnf(args, workspace, repositories, base_packages, boot_packages, conf
     with mount_api_vfs(args, workspace):
         run(cmdline, check=True)
 
+@complete_step('Installing Clear Linux')
+def install_clear(args, workspace, run_build_script):
+    if args.release == "latest":
+        release = "clear"
+    else:
+        release = "clear/"+args.release
+
+    root = os.path.join(workspace, "root")
+
+    packages = ['os-core'] + args.packages
+    if run_build_script:
+        packages.extend(args.build_packages)
+    if args.bootable:
+        packages += ['kernel-native']
+
+    # Since swupd-extract program is usually installed with "go get",
+    # try to find it in some other locations besides PATH.
+    swupd_extract = None
+    sudo_user = os.getenv("SUDO_USER")
+    if sudo_user:
+        sudo_user_program = os.path.expanduser("~{}/go/bin/swupd-extract".format(sudo_user))
+        if os.path.exists(sudo_user_program):
+            swupd_extract = sudo_user_program
```

Comment:
uh, oh, this sudo stuff here is … wild …

are we sure we want this bit? That just appears too magic to me...

Reply:
The path is standard for go programs. I find it pretty convenient for not having to edit PATH for a single go program I have on my machine. Maybe move this to a `def find_go_program()`?

Reply:
hmpf, i find automatism that reach across user identity boundaries like this a bit too magic and dangerous for my tastes... Can't we find a different solution for this? Maybe add a new setting to mkosi.default that basically alters the search paths we use for all our tools? think:

```
ExtraSearchPath=$HOME/go/bin:$SUDO_HOME/go/bin
```

That way, you could make things work for your case automatically, but we wouldn't have to hardcode this into the normal logic. And it would be generically useful, and could cover all other external tools we need.

Of course, `$SUDO_HOME` isn't actually defined by sudo currently, but I think it would be fine to fake it in in our code if it's not set. I mean, faking such env vars isn't conceptually a new thing, given the xdg basedir stuff which in theory one day we could support the same way...

What do you think?

Reply:
and of course, ExtraSearchPath= should be somewhat "smart", i.e. first split the specified string at colons, and then replace the env vars, and if any env vars can't be replaced because they are not set currently, and not faked either, then skip the whole item. i.e. be smarter than what the shell would do in this case, where it would replace unset env vars by the empty string...

Reply:
humn... that looks good to me. @cmarcelo ?